dot/rpc/modules/grandpa.go

```diff
@@ -71,3 +97,101 @@ func (gm *GrandpaModule) ProveFinality(r *http.Request, req *ProveFinalityReques
 
 	return nil
 }
+
+// RoundState returns the state of the current best round state as well as the ongoing background rounds.
+func (gm *GrandpaModule) RoundState(r *http.Request, req *EmptyRequest, res *RoundStateResponse) error {
+	voters := gm.blockFinalityAPI.GetVoters()
+	votersPkBytes := make([]ed25519.PublicKeyBytes, len(voters))
+	for i, v := range voters {
+		votersPkBytes[i] = v.PublicKeyBytes()
+	}
+
+	prevotes := gm.blockFinalityAPI.PreVotes()
+	precommits := gm.blockFinalityAPI.PreCommits()
+
+	missingPrevotes, err := toAddress(difference(votersPkBytes, prevotes))
+	if err != nil {
+		return err
+	}
+
+	missingPrecommits, err := toAddress(difference(votersPkBytes, precommits))
+	if err != nil {
+		return err
+	}
+
+	totalWeight := calcTotalWeight(voters)
+
+	roundstate := RoundStateResponse{
+		SetID: uint32(gm.blockFinalityAPI.GetSetID()),
+		Best: RoundState{
+			Round:           uint32(gm.blockFinalityAPI.GetRound()),
+			ThresholdWeight: calcThresholdWeight(totalWeight),
+			TotalWeight:     totalWeight,
+			Prevotes: Votes{
+				CurrentWeight: calcWeight(voters, prevotes),
+				Missing:       missingPrevotes,
+			},
+			Precommits: Votes{
+				CurrentWeight: calcWeight(voters, precommits),
+				Missing:       missingPrecommits,
+			},
+		},
+		Background: []RoundState{},
```

Comment:
Can we use `nil` here?

Reply:
It's my fault, these changes are not related to this PR, I've already removed and kept just the necessary modifications. Same to the other change